lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -0,0 +1,129 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Task::Git::Clone;
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
+
+use OpenQA::Utils qw(run_cmd_with_log_return_error);
+use Cwd 'abs_path';
+
+sub register ($self, $app, @) {
+    $app->minion->add_task(git_clone => \&_git_clone);
+}
+
+sub _git_clone ($job, $clones) {
+    # $clones is a hashref with paths as keys and urls to git repos as values.
+    # The urls may also refer to a branch via the url fragment.
+   # If no branch is set, the default branch of the remote (if target path doesn't exist yet) or local checkout is used.
+# If the target path already exists, an update of the specified branch is performed, if the url matches the origin remote
+
+    my $app = $job->app;
+    my $job_id = $job->id;
+
+    # Prevent multiple git clone tasks for the same path to run in parallel
+    my @guards;
+    for my $path (sort keys %$clones) {
+        while (-l $path) { $path = abs_path($path); }    # guard symlink target
+        my $guard = $app->minion->guard("limit_git_clone_${path}_task", 7200);
+        return $job->retry({delay => 30 + int(rand(10))}) unless $guard;
+        push(@guards, $guard);
+    }
+
+    my $log = $app->log;
+    my $ctx = $log->context("[#$job_id]");
+
+    # iterate clones sorted by path length to ensure that a needle dir is always cloned after the corresponding casedir
+    for my $path (sort { length($a) > length($b) } keys %$clones) {
+        my $url = $clones->{$path};
+        die("Aber echt ned!") if $path =~ /\.\./;
```

Comment:
I guess we will eventually have to put a more serious error message there.

Of course this breaks if one has simply two `..` in the middle of a path element but I guess we don't care.

Reply:
better? ;)